test/remote_partition/remote_part_test.js

```diff
@@ -0,0 +1,104 @@
+var should = require('should');
+var Tapeworm = require('tapeworm');
+var Commit = require('tapeworm').Commit;
+var SocketMock = require('socket.io-mock');
+var TapewormSyncServer = require('../../lib/server');
+var RemoteDriver = require('../../lib/client').RemotePersistence;
+var Client = require('../../lib/client').RemoteClient;
+
+var autobus = {
+  leave: function () {}
+};
+
+describe('Remote partition server/client', function () {
+  var socketServer, socketClient, clientTapeworm, tapeworm, serverPartition = null, clientPartition = null;
+
+  beforeEach(function (done) {
+    socketServer = new SocketMock();
+    socketClient = socketServer.socketClient;
+    var rrtwClient = new Client(socketClient, '/tw');
+    var remotePartition = new RemoteDriver(rrtwClient, 'blondie');
+
+    clientTapeworm = new Tapeworm(remotePartition);
+    tapeworm = new Tapeworm();
+    tapeworm.openPartition('blondie').then(function(blondie) {
+      serverPartition = blondie;
+      clientTapeworm.openPartition('blondie').then(function (blondie2) {
+        clientPartition = blondie2;
+        done();
+      });
+    });
+  });
+
+  it('#socketMock should work', function (done) {
+    socketClient.on('ping', function (message) {
+      message.should.equal('Hello');
+      socketClient.emit('pong', 'heya');
+    });
+    socketServer.on('pong', function (message) {
+      message.should.equal('heya');
+      done();
+    });
+    socketServer.emit('ping', 'Hello');
+  });
+
+  it('#loadSnapshot should work on remote partition', function (done) {
+    const tapewormSyncServer = new TapewormSyncServer(serverPartition, autobus);
+    serverPartition.storeSnapshot('1', { test: 'success' }, 2, function () {
+      clientPartition.loadSnapshot('1', function (err, res) {
+        res.snapshot.test.should.equal('success');
+        done();
+      });
+    });
+    tapewormSyncServer.addSocket(socketServer);
+  });
+  it('#loadSnapshot should work on remote partition with promise', function (done) {
```

Comment:
Best would maybe be to change the interface to do snapshot and query in the same call?